File: api_service/models/thread.py
from datetime import datetime
from typing import Optional

import orjson
from pydantic import BaseModel

from models.common import orjson_dumps


class Thread(BaseModel):
    id: int = 0
    slug: Optional[str] = None
    title: str
    message: str
    created: datetime = datetime.now()
    author: str
    forum: str = ''
    votes: int = 0

    class Config:
        json_loads = orjson.loads
        json_dumps = orjson_dumps


class ThreadUpdate(BaseModel):
    title: str = ''
    message: str = ''

    class Config:
        json_loads = orjson.loads
        json_dumps = orjson_dumps

File: api_service/models/thread.go
package models

import "time"

//easyjson:json
type Thread struct {
	Id      int64     `json:"id,omitempty"`
	Slug    string    `json:"slug,omitempty"`
	Title   string    `json:"title,omitempty"`
	Message string    `json:"message,omitempty"`
	Created time.Time `json:"created,omitempty"`
	Author  string    `json:"author,omitempty"`
	Forum   string    `json:"forum,omitempty"`
	Votes   int       `json:"votes,omitempty"`
}

//easyjson:json
type ThreadUpdate struct {
	Title   string `json:"title,omitempty"`
	Message string `json:"message,omitempty"`
}
